src/subcommand/fix.rs

```diff
@@ -270,16 +270,33 @@ impl Error {
     }
 }
 
+/// A logger. This is used to print the output to stdout
+pub trait Logger {
+    /// Log things. This is mutable because TestLogger mutates when we log
+    fn log(&mut self, content: &str);
+}
+
+/// A logger that just prints to stdout
+pub struct StdoutLogger;
+
+impl Logger for StdoutLogger {
+    fn log(&mut self, content: &str) {
+        println!("{content}");
+    }
+}
+
 /// The primary entrypoint for the fix command.
 #[tracing::instrument(skip_all, fields(subcommand = "fix"))]
-pub async fn main(config: &Config) -> Result<(), Report<Error>> {
-    let integration_errors = check_integrations(config).await;
-    let fossa_connection_errors = check_fossa_connection(config).await;
+pub async fn main(config: &Config, logger: &mut dyn Logger) -> Result<(), Report<Error>> {
```

Comment:
By using `dyn`, you're forcing this method to use [dynamic dispatch](https://doc.rust-lang.org/std/keyword.dyn.html).

Instead, prefer to accept a generic type that can be statically dispatched:
```suggestion
pub async fn main<L: Logger>(config: &Config, logger: &L) -> Result<(), Report<Error>> {
```

Alternately, you can use the below syntax to mean the same thing:
```rust
pub async fn main(config: &Config, logger: &impl Logger) -> Result<(), Report<Error>> {
```

Commenting just once here, but please address generally.